tests/foreman/ui/test_classparameters.py

```diff
@@ -0,0 +1,469 @@
+        """Override the Default Parameter value.
+
+        @feature: Class Parameters - Override
+
+        @steps:
+
+        1.  Check the Override checkbox.
+        2.  Set the new valid Default Value.
+        3.  Submit the changes.
+
+        @assert: Parameter Value overridden with new value.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_negative_override_checkbox(self):
+        """Override the Default Parameter value - override Unchecked.
+
+        @feature: Class Parameters - Override
+
+        @steps:
+
+        1.  Don't check the Override checkbox.
+        2.  Set the new valid Default Value.
+        3.  Attempt to submit the changes.
+
+        @assert: Parameter value not allowed/disabled to override.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_edit_parameter_dialog(self):
+        """Enable Validation, merging and matcher sections.
+
+        @feature: Class Parameters - Edit Dialog
+
+        @steps:
+
+        1.  Check the Override checkbox.
+
+        @assert: Puppet Default, Hiding, Validation, Merging and
+        Matcher section enabled.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_negative_edit_parameter_dialog(self):
+        """Disable Validation, merging and matcher sections.
+
+        @feature: Class Parameters - Edit Dialog
+
+        @steps:
+
+        1.  Dont't Check the Override checkbox.
+
+        @assert: Puppet Default, Hiding, Validation, Merging and
+        Matcher section is disabled.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_puppet_default(self):
+        """On Override, Set Puppet Default Value.
+
+        @feature: Class Parameters - Puppet Default
+
+        @steps:
+
+        1.  Check the Override checkbox.
+        2.  Check 'Use Puppet Default' checkbox.
+        3.  Submit the changes.
+
+        @assert: Puppet Default Value applied on parameter.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_update_parameter_type(self):
+        """Positive Parameter Update for parameter types - Valid Value.
+
+        Types: string, boolean, integer, real, array, hash, yaml, json
+
+        @feature: Class Parameters - Parameter Type
+
+        @steps:
+
+        1.  Check the Override checkbox.
+        2.  Update the Key Type.
+        3.  Enter a 'valid' default Value.
+        3.  Submit the changes.
+
+        @assert: Parameter Updated with a new type successfully.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_update_parameter_type(self):
+        """Negative Parameter Update for parameter types - Invalid Value.
+
+        Types: string, boolean, integer, real, array, hash, yaml, json
```

Comment:
Testimony currently doesn't allow : in test case names. Please

s/'Types:'/'Types - '